File: go/orm/sample/migrations/Table/UserRole/V0005_AddGrantedBy.go
package userrole

import (
	"github.com/kosmassp/SimpleOrm/go/orm"
	"github.com/kosmassp/SimpleOrm/go/orm/sample"
)

// V0005_AddGrantedBy adds an optional column recording who granted the role.
type V0005_AddGrantedBy struct {
	orm.TableMigration[sample.UserRole]
}

func (V0005_AddGrantedBy) Action(actions *orm.TableActions) {
	actions.AddColumn("granted_by", "TEXT")
}

File: php/samples/SimpleOrm.Sample/src/Migrations/Table/UserRole/V0005_AddGrantedBy.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Sample\Migrations\Table\UserRole;

use SimpleOrm\Migrations\TableActions;
use SimpleOrm\Migrations\TableMigration;
use SimpleOrm\Sample\Models\Tables\UserRole;

final class V0005_AddGrantedBy extends TableMigration
{
    public function entityClass(): string
    {
        return UserRole::class;
    }

    public function action(TableActions $actions): void
    {
        $actions->addColumn('granted_by', 'TEXT');
    }
}
